dev_scripts/exportTranslation.js

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable default-case */
+/* eslint-disable array-callback-return */
+/* eslint-disable func-names */
+/* eslint-disable import/no-dynamic-require */
+/* eslint-disable global-require */
+/* eslint-disable no-console */
+/* eslint-disable no-useless-escape */
+/* eslint-disable no-shadow */
+/* eslint-disable guard-for-in */
+/* eslint-disable prefer-const */
+/* eslint-disable no-undef */
+/* eslint-disable no-restricted-syntax */
+
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+const fs = require('fs');
+
+const localizationFiles = [];
+let index = 0;
+
+// Define mainLanguage as English (gb)
+const mainLanguage = 'gb';
+
+getLocalizationFiles = () => {
+  // Reads all files in localization files and save them in localizationFiles array
+  fs.readdirSync('././src/localization/').forEach(file => {
+    // Filter to only get JSON files
+    if (file.split('.').pop() === 'json') {
+      const fileName = require(`../src/localization/${file}`);
+      localizationFiles.push(fileName);
+      localizationFiles[index].name = file.replace('.json', '');
+      index += 1;
+    }
+  });
+};
+
+// Gets the language entered by the User (fr, la, gil, tl)
+getSelectedLanguage = () => process.argv[2];
+
+generateCsvData = (mainLanguage, selectedLanguage) => {
+  let valueFound = false;
+  // csvFileData keeps all the data stored in json files
+  let csvData = '';
+  // Goes through all localization files
+  localizationFiles.map(localizationFile => {
+    const mainLanguageKeys = [];
+    const mainLanguageValues = [];
+    const selectedLanguageValues = [];
+    // Iterates over every language
+    for (const [language, mainLanguageKey] of Object.entries(localizationFile)) {
```

Comment:
translation scripts only allow for one language to be exports with one mainLanguage, there is no need for this for loop or the case statement:

You can quite easily generate the arrays you use with in one for loop, iterating through Object.entries(localizationFile[mainLanguage) and while adding to mainLanguageKeys and mainLanguageValues you can check if localizationFile[selectedLanguage][mainLanguageKeyAgain] exists and add appropriate value to selectedLanguageValues.

I am not a big fan of how you've named the variables btw, `mainLanguageKey` and `mainLanguageKeys` and `mainLanguageKeyAgain` are not very representative of their function

Reply:
Good advice, after some changes I really did not use some of them and I just directly replaced `mainLanguage` for `gb` as we know we are talking about english.